zds/searchv2/models.py

```diff
@@ -6,73 +6,69 @@
 from django.db import models
 from django.conf import settings
 
-from elasticsearch.helpers import parallel_bulk
-from elasticsearch import ConnectionError
-from elasticsearch_dsl import Mapping
-from elasticsearch_dsl.query import MatchAll
-from elasticsearch_dsl.connections import connections
-
 from django.db import transaction
+from typesense import Client
 
 
-def es_document_mapper(force_reindexing, index, obj):
+def document_indexer(force_reindexing, obj):
     action = "update" if obj.es_already_indexed and not force_reindexing else "index"
-    return obj.get_es_document_as_bulk_action(index, action)
+    return obj.get_document_as_bulk_action(action)
 
 
-class AbstractESIndexable:
+class AbstractSearchIndexable:
     """Mixin for indexable objects.
 
     Define a number of different functions that can be overridden to tune the behavior of indexing into elasticsearch.
 
     You (may) need to override :
 
     - ``get_indexable()`` ;
-    - ``get_mapping()`` (not mandatory, but otherwise, ES will choose the mapping by itself) ;
-    - ``get_document()`` (not mandatory, but may be useful if data differ from mapping or extra stuffs need to be done).
+    - ``get_schema()`` (not mandatory, but otherwise, ES will choose the schema by itself) ;
+    - ``get_document()`` (not mandatory, but may be useful if data differ from schema or extra stuffs need to be done).
 
-    You also need to maintain ``es_id`` and ``es_already_indexed`` for bulk indexing/updating (if any).
+    You also need to maintain ``index_id`` and ``es_already_indexed`` for bulk indexing/updating (if any).
     """
 
     es_already_indexed = False
-    es_id = ""
+    index_id = ""
```

Comment:
```suggestion
    search_engine_id = ""
```
me paraît mieux, car sinon on peut ne pas savoir de quel index il s'agit.